src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -307,13 +321,21 @@ bool CartGridStrategy::load_grid(StreamOutput *stream)
 
     for (int y = 0; y < configured_grid_y_size; y++) {
         for (int x = 0; x < configured_grid_x_size; x++) {
-            if(fread(&grid[x + (configured_grid_x_size * y)], sizeof(float), 1, fp) != 1) {
+            float read_value;
+            if(fread(&read_value , sizeof(float), 1, fp) != 1) {
                 stream->printf("error:Failed to read grid\n");
                 fclose(fp);
                 return false;
+            } else {
+                if (read_value != grid[x + (configured_grid_x_size * y)]) {
+                    grid[x + (configured_grid_x_size * y)] = read_value;
```

Comment:
This optimizes nothing it is RAM it makes no difference if it is written to or not. This is not an optimization IMO, it just complicates the code and adds an extra variable.

Reply:
Again... not RAM.  There was not enough RAM to implement the function so we allocated the block in program memory.  It has limited write cycles, and some of my older systems are starting to fail randomly especially in the z leveling

Reply:
May I suggest you dig out the LPC1768 manual and read it ;)